File: solutions/scala/grade-school/4/src/main/scala/GradeSchool.scala
class School:
  type DB = Map[Int, Seq[String]]
  private var database: DB = Map()

  def add(name: String, g: Int) =
    val updatedGrade = (database.getOrElse(g, Seq()) :+ name)
    database = database.updated(g, updatedGrade)

  def db: DB = database

  def grade(g: Int): Seq[String] = database.getOrElse(g, Seq())

  def sorted: DB =
    database
      .transform((_, students) => students.sorted)
      .toSeq
      .sortBy(_._1)
      .toMap

File: solutions/clojure/grade-school/3/src/grade_school.clj
(ns grade-school)

(defn grade [school grade]
  (school grade []))

(defn add [school name grade]
  (assoc school grade (conj (school grade []) name)))

(defn sorted [school]
  (into (sorted-map) (for [[k v] school] [k (sort v)])))